Tests/FunctionalTests/TestDiscoveryTests.swift

```diff
@@ -12,159 +12,185 @@
 
 import Basics
 import PackageModel
+import SPMBuildCore
 import _InternalTestSupport
-import XCTest
+import Testing
+import struct Foundation.UUID
+import class Foundation.ProcessInfo
 
-final class TestDiscoveryTests: XCTestCase {
-    func testBuild() async throws {
-        try await fixture(name: "Miscellaneous/TestDiscovery/Simple") { fixturePath in
-            let (stdout, _) = try await executeSwiftBuild(fixturePath)
-            // in "swift build" build output goes to stdout
-            XCTAssertMatch(stdout, .contains("Build complete!"))
+@Suite
+struct TestDiscoveryTests {
+    static var buildSystems: [BuildSystemProvider.Kind] = [BuildSystemProvider.Kind.native, .swiftbuild]
+
+    @Test(arguments: buildSystems)
```

Comment:
suggestion (non-blocking, if-pr-update):  using `arguments: SupportedBuildSystemOnAllPlatforms` does essentially the same thing :)